### ISSUE TITLE
Changing Thermostat Operation Mode broken

### ISSUE THREAD

**Issue by philhawthorne** [pavoni/pyvera#41]
This is a followup to home-assistant/home-assistant#5828

Changing the Operation Mode for Thermostats seems to be broken. To ensure we're all on the same page I'm using Home Assistant 0.39.1, however this issue was first reported in Home Assistant 0.37.

I've increased the logging level, to see what is going on when changing the Operation Mode for my Living Room thermostat. For reference, it is a Remotec ZXT-120 IR controller paired with a Vera Lite. It works in Home Assistant 0.36.

**Off -> Cool**
```
17-02-27 20:43:31 DEBUG (Thread-10) [pyvera] Result of vera_request with payload {'id': 'lu_action', 'serviceId': ('urn:upnp-org:serviceId:HVAC_UserOperatingMode1',), 'action': 'SetSetModeTarget', 'NewModeTarget': 'CoolOn'}: ERROR: Device does not handle service/action
```
No action is taken from the ZXT-120

**Device State Off - Setting Temp to 22**
```
17-02-27 20:46:04 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service_data=temperature=34, entity_id=climate.living_room_ac, service=set_temperature, domain=climate, service_call_id=140218452162264-7>
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Set variable: CurrentSetpoint using service_id: urn:upnp-org:serviceId:TemperatureSetpoint1
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Result of vera_request with payload {'id': 'lu_action', 'serviceId': 'urn:upnp-org:serviceId:TemperatureSetpoint1', 'action': 'SetCurrentSetpoint', 'NewCurrentSetpoint': 34.0}: { "u:SetCurrentSetpointResponse": { "JobID": "17875" } }
17-02-27 20:46:04 INFO (Vera Poll Thread) [pyvera.subscribe] Poll returned
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Set variable: CurrentSetpoint using service_id: urn:upnp-org:serviceId:TemperatureSetpoint1_Heat
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Result of vera_request with payload {'id': 'lu_action', 'serviceId': 'urn:upnp-org:serviceId:TemperatureSetpoint1_Heat', 'action': 'SetCurrentSetpoint', 'NewCurrentSetpoint': 34.0}: { "u:SetCurrentSetpointResponse": { "JobID": "17876" } }
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Set variable: CurrentSetpoint using service_id: urn:upnp-org:serviceId:TemperatureSetpoint1_Cool
17-02-27 20:46:04 DEBUG (Thread-12) [pyvera] Result of vera_request with payload {'id': 'lu_action', 'serviceId': 'urn:upnp-org:serviceId:TemperatureSetpoint1_Cool', 'action': 'SetCurrentSetpoint', 'NewCurrentSetpoint': 34.0}: { "u:SetCurrentSetpointResponse": { "JobID": "17877" } }
17-02-27 20:46:04 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]: service_call_id=140218452162264-7>
17-02-27 20:46:04 DEBUG (MainThread) [homeassistant.components.websocket_api] WS 140217463741184: Sending {'id': 12, 'type': 'result', 'success': True, 'result': None}
17-02-27 20:46:04 INFO (Thread-7) [homeassistant.components.device_tracker.nmap_tracker] Scanning...
17-02-27 20:46:05 INFO (Vera Poll Thread) [pyvera.subscribe] Poll returned
17-02-27 20:46:05 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=climate.living_room_ac, old_state=<state climate.living_room_ac=Off; current_temperature=30, temperature=34.5, friendly_name=Living Room AC, fan_mode=Auto, battery_level=100%, unit_of_measurement=°C, operation_mode=Off, Vera Device Id=18, fan_list=['On', 'Auto', 'Cycle'], max_temp=35, min_temp=7, operation_list=['Heat', 'Cool', 'Auto Changeover', 'Off'] @ 2017-02-27T20:41:42.372612+11:00>, new_state=<state climate.living_room_ac=Off; current_temperature=30, temperature=34.0, friendly_name=Living Room AC, fan_mode=Auto, battery_level=100%, unit_of_measurement=°C, operation_mode=Off, Vera Device Id=18, fan_list=['On', 'Auto', 'Cycle'], max_temp=35, min_temp=7, operation_list=['Heat', 'Cool', 'Auto Changeover', 'Off'] @ 2017-02-27T20:41:42.372612+11:00>>
17-02-27 20:46:06 DEBUG (MainThread) [homeassistant.components.websocket_api] WS 140217463741184: Sending {'id': 2, 'type': 'event', 'event': {'time_fired': datetime.datetime(2017, 2, 27, 9, 46, 5, 965904, tzinfo=<UTC>), 'event_type': 'state_changed', 'data': {'entity_id': 'climate.living_room_ac', 'old_state': <state climate.living_room_ac=Off; current_temperature=30, temperature=34.5, friendly_name=Living Room AC, fan_mode=Auto, battery_level=100%, unit_of_measurement=°C, operation_mode=Off, Vera Device Id=18, fan_list=['On', 'Auto', 'Cycle'], max_temp=35, min_temp=7, operation_list=['Heat', 'Cool', 'Auto Changeover', 'Off'] @ 2017-02-27T20:41:42.372612+11:00>, 'new_state': <state climate.living_room_ac=Off; current_temperature=30, temperature=34.0, friendly_name=Living Room AC, fan_mode=Auto, battery_level=100%, unit_of_measurement=°C, operation_mode=Off, Vera Device Id=18, fan_list=['On', 'Auto', 'Cycle'], max_temp=35, min_temp=7, operation_list=['Heat', 'Cool', 'Auto Changeover', 'Off'] @ 2017-02-27T20:41:42.372612+11:00>}, 'origin': 'LOCAL'}}
```
No action is taken by the ZXT-120

**Off -> Heat**
```
17-02-27 20:47:44 DEBUG (MainThread) [homeassistant.components.websocket_api] WS 140217463741184: Received {'service_data': {'entity_id': 'climate.living_room_ac', 'operation_mode': 'Heat'}, 'type': 'call_service', 'service': 'set_operation_mode', 'domain': 'climate', 'id': 13}
17-02-27 20:47:44 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: service_data=entity_id=climate.living_room_ac, operation_mode=Heat, service=set_operation_mode, domain=climate, service_call_id=140218452162264-8>
17-02-27 20:47:45 DEBUG (Thread-1) [pyvera] Result of vera_request with payload {'id': 'lu_action', 'serviceId': ('urn:upnp-org:serviceId:HVAC_UserOperatingMode1',), 'action': 'SetSetModeTarget', 'NewModeTarget': 'HeatOn'}: ERROR: Device does not handle service/action
17-02-27 20:47:45 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]: service_call_id=140218452162264-8>
17-02-27 20:47:45 DEBUG (MainThread) [homeassistant.components.websocket_api] WS 140217463741184: Sending {'id': 13, 'type': 'result', 'success': True, 'result': None}
17-02-27 20:47:48 INFO (Thread-3) [homeassistant.components.device_tracker.nmap_tracker] Scanning...
17-02-27 20:47:49 DEBUG (Thread-5) [homeassistant.components.device_tracker.bluetooth_tracker] Bluetooth devices discovered = 0
17-02-27 20:47:49 DEBUG (Thread-5) [homeassistant.components.device_tracker.bluetooth_tracker] Scanning AC:37:43:79:4D:C1
17-02-27 20:47:50 DEBUG (Thread-5) [homeassistant.components.device_tracker.bluetooth_tracker] Scanning 60:92:17:AF:11:E3
17-02-27 20:47:50 INFO (Thread-3) [homeassistant.components.device_tracker.nmap_tracker] nmap scan successful
17-02-27 20:47:53 INFO (Vera Poll Thread) [pyvera.subscribe] Poll returned
```
No action is taken by the ZXT-120

So it seems there's an issue with whatever is throwing `ERROR: Device does not handle service/action`. 

Hope this helps!

**Comment by pavoni** [comment]
Vera helpful - `SetSetModeTarget ` has too many `Set`s!

Hopefully fixed in the latest release. If you can easily try a dev branch - can you checkout my branch `fix_vera_thormostat_bug` (excuse the typo - typed the branch name too fast!)

**Comment by pavoni** [comment]
Fixed in `0.2.24`
